File: 1-twoSum/twoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        obj = {}
        for i in range(0, len(nums)):
            remainder = target - nums[i]
            if remainder in obj:
                return [i, obj[remainder]]
            obj[nums[i]] = i

File: 1-twoSum/twoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
      HashMap<Integer, Integer> obj = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
          int remainder = target - nums[i];
          if (obj.get(remainder) != null) {
            int[] res = { i, obj.get(remainder)};
            return res;
          }
          obj.put(nums[i], i);
        }
        return null;
    }
}